lang/rust/avro/src/schema.rs

```diff
@@ -335,6 +335,25 @@ impl fmt::Display for Name {
     }
 }
 
+impl<'de> Deserialize<'de> for Name {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        serde_json::Value::deserialize(deserializer).and_then(|value| {
+            use serde::de::Error;
+            if let Value::Object(json) = value {
+                Name::parse(&json).map_err(D::Error::custom)
```

Comment:
Would there ever be a situation where this is used to parse a JSON Value of a string? Probably not because full schema parsing will always probably be routed through the parser I imagine?